base_layer/core/src/mempool/service/inbound_handlers.rs

```diff
@@ -58,9 +70,30 @@ where T: BlockchainBackend
     }
 
     /// Handle inbound transactions from remote wallets and local services.
-    pub async fn handle_transaction(&self, tx: &Transaction) -> Result<(), MempoolServiceError> {
-        // TODO tx must pass through the validation pipeline, with checking of its internal consistency, before adding
-        // and propagating it.
-        Ok(self.mempool.insert(Arc::new(tx.clone()))?)
+    pub async fn handle_transaction(
+        &mut self,
+        tx: &Transaction,
+        source_peer: Option<CommsPublicKey>,
+    ) -> Result<(), MempoolServiceError>
+    {
+        let add_result = self.mempool.insert(Arc::new(tx.clone()));
+        let exclude_list = if let Some(peer) = source_peer {
+            vec![peer]
+        } else {
+            Vec::new()
+        };
```

Comment:
Shouldn't this be inside the `if` on L85?

Reply:
I agree, it would be better inside